contracts/secret-storage/SecretManager.sol

```diff
@@ -356,90 +356,72 @@ contract SecretManager is
     }
 
     function _markStoreDead(
-        uint256 _secretId
+        address _enclaveAddress,
+        uint256[] memory _secretIds
     ) internal {
-        for (uint256 index = 0; index < userStorage[_secretId].selectedEnclaves.length; ) {
-            bool isArrayLengthReduced = _markEnclaveDead(_secretId, userStorage[_secretId].selectedEnclaves[index].enclaveAddress, index);
-            if(!isArrayLengthReduced)
-                ++index;
+        // FIX: get store data
+        ( , , , uint256 lastAliveTimestamp, , ) = SECRET_STORE.secretStorage(_enclaveAddress);
+        if(block.timestamp <= lastAliveTimestamp + MARK_ALIVE_TIMEOUT)
+            revert SecretManagerStoreIsAlive();
+
+        for (uint256 index = 0; index < _secretIds.length; index++) {
+            uint256 enclaveIndex = _getSelectedEnclaveIndex(_secretIds[index], _enclaveAddress);
+            _markEnclaveDead(_secretIds[index], _enclaveAddress, enclaveIndex);
         }
 
-        // TODO: delete data and refund remaining
-        if(block.timestamp > userStorage[_secretId].endTimestamp && userStorage[_secretId].selectedEnclaves.length == 0)
-            _refundExcessDepositAndRemoveStore(_secretId);
+        SECRET_STORE.updateLastAliveTimestamp(_enclaveAddress, block.timestamp);
     }
 
     function _markEnclaveDead(
         uint256 _secretId,
         address _enclaveAddress,
         uint256 _enclaveIndex
-    ) internal returns (bool isArrayLengthReduced) {
+    ) internal {
         if(!userStorage[_secretId].selectedEnclaves[_enclaveIndex].hasAcknowledgedStore)
-            return isArrayLengthReduced;
+            return;
 
-        if(block.timestamp <= userStorage[_secretId].selectedEnclaves[_enclaveIndex].lastAliveTimestamp + MARK_ALIVE_TIMEOUT)
-            return isArrayLengthReduced;
+        uint256 noOfSelectedEnclaves = userStorage[_secretId].selectedEnclaves.length;
+        _updateUsdcDeposit(_secretId, noOfSelectedEnclaves);
 
         // case for when the termination condition is reached, we won't select any new enclave
         if(block.timestamp > userStorage[_secretId].endTimestamp) {
-            isArrayLengthReduced = true;
             _removeSelectedEnclave(_secretId, _enclaveIndex);
+            if(userStorage[_secretId].selectedEnclaves.length == 0)
+                _refundExcessDepositAndRemoveStore(_secretId);
         }
         // case for when the termination condition is pending, we will try to select a new enclave in place of the dead enclave
         else {
             SelectedEnclave[] memory selectedEnclaves = SECRET_STORE.selectEnclaves(1, userStorage[_secretId].sizeLimit);
             // case for when a new enclave can't be selected as they all are already occupied to their max storage capacity
             if (selectedEnclaves.length == 0) {
-                isArrayLengthReduced = true;
                 _removeSelectedEnclave(_secretId, _enclaveIndex);
                 emit SecretResourceUnavailable(_secretId);
```

Comment:
note: will update this event name `SecretReplicationReduced(id, replication_count)`